js/app/test/audio_component_events.spec.ts

```diff
@@ -69,28 +69,26 @@ test("Play, Pause, and stop events work correctly.", async ({ page }) => {
 	const uploader = await page.locator("input[type=file]");
 	await Promise.all([
 		uploader.setInputFiles(["../../demo/audio_debugger/cantina.wav"]),
-		page.waitForResponse("**/upload?*")
+		page.waitForResponse("**/upload*?*")
 	]);
 
+	const event_triggered = async (label: string) => {
+		const value = await page.getByLabel(label).inputValue();
+		expect(Number(value)).toBeGreaterThan(0);
+	};
+
 	await page
 		.getByTestId("waveform-Output Audio")
 		.getByLabel("Play", { exact: true })
 		.click();
+	await expect(async () => event_triggered("# Output Play Events")).toPass();
```

Comment:
I think what's happening is that we're clicking the pause button immediately after the play button and I think the server is not reacting as fast as the playwright client.

So what I did is wait for the play event to finish triggering. Then trigger the pause event. Same for video.

Reply:
Nice.